internal/util/search/pathparser.go

```diff
@@ -10,17 +10,61 @@ func (sr *SearchReplace) pathMatch(yamlPath string) bool {
 	if sr.ByPath == "" {
 		return false
 	}
-	inputElems := strings.Split(sr.ByPath, PathDelimiter)
-	traversedElems := strings.Split(strings.Trim(yamlPath, PathDelimiter), PathDelimiter)
-	if len(inputElems) != len(traversedElems) {
-		return false
+
+	patternElems := strings.Split(sr.ByPath, PathDelimiter)
+	yamlPathElems := strings.Split(strings.TrimPrefix(yamlPath, PathDelimiter), PathDelimiter)
+	return backTrackMatch(yamlPathElems, patternElems)
+}
+
+// backTrackMatch matches the traversed yamlPathElems with input(from by-path) patternElems
+// * matches any element, ** matches 0 or more elements, array elements are split and matched
+// refer to pathparser_test.go
+func backTrackMatch(yamlPathElems, patternElems []string) bool {
+	yamlPathElemsLen, patternElemsLen := len(yamlPathElems), len(patternElems)
```

Comment:
Can we explain with some comments the overall idea about what we are doing in the function? It is pretty hard to follow.

Reply:
Done. This is similar to wildcard match https://leetcode.com/problems/wildcard-matching/